src/packages/sdk/src/engine-commands/getDmmf.ts

```diff
@@ -0,0 +1,192 @@
+import Debug from '@prisma/debug'
+import { NApiEngineTypes } from '@prisma/engine-core'
+import { EngineTypes } from '@prisma/fetch-engine'
+import { DataSource, DMMF, GeneratorConfig } from '@prisma/generator-helper'
+import chalk from 'chalk'
+import execa from 'execa'
+import fs from 'fs'
+import tmpWrite from 'temp-write'
+import { promisify } from 'util'
+import { resolveBinary } from '../resolveBinary'
+
+const debug = Debug('prisma:getDMMF')
+
+const unlink = promisify(fs.unlink)
+
+const MAX_BUFFER = 1_000_000_000
+
+export interface ConfigMetaFormat {
+  datasources: DataSource[]
+  generators: GeneratorConfig[]
+  warnings: string[]
+}
+
+export type GetDMMFOptions = {
+  datamodel?: string
+  cwd?: string
+  prismaPath?: string
+  datamodelPath?: string
+  retry?: number
+  previewFeatures?: string[]
+}
+
+export async function getDMMF({
+  datamodel,
+  cwd = process.cwd(),
+  prismaPath: queryEnginePath,
+  datamodelPath,
+  retry = 4,
+  previewFeatures,
+}: GetDMMFOptions): Promise<DMMF.Document> {
+  warnOnDeprecatedFeatureFlag(previewFeatures)
+
+  const useNapi =
+    process.env.PRISMA_FORCE_NAPI === 'true' ||
+    previewFeatures?.includes('nApi')
+  if (useNapi) {
+    queryEnginePath = await resolveBinary(
```

Comment:
I would love for this file code to be chopped up into functions, it would help me understand your intent better

Reply:
Implemented